### PR TITLE
chore: [SVLS-5265] add span links to encoded spans

### PR DESCRIPTION
**Pull request by apiarian-datadog**
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by apiarian-datadog** [comment]
followup from https://github.com/DataDog/datadog-lambda-python/pull/513 where we are using the logwriter output and missing the span links for our integration tests.

File: tests/integration/test_encoding.py [hunk 1]
```diff
@@ -63,3 +63,22 @@ def test_trace_with_metrics_accepted_by_agent(self, metrics):
...
         log.error.assert_not_called()
+
+    @pytest.mark.parametrize(
+        "span_links_kwargs",
+        [
+            {"trace_id": 12345, "span_id": 67890},
+        ],
+    )
+    def test_trace_with_links_accepted_by_agent(self, span_links_kwargs):
+        """Links should not break things."""
```
Comment (ZStriker19) [inline tests/integration/test_encoding.py:74]:
```suggestion
        """The agent does not return an error code when accepting a trace payload containing span links."""
```

**Review by ZStriker19 — APPROVED** [review]

File: tests/integration/test_encoding.py [hunk 2]
```diff
@@ -63,3 +63,22 @@ def test_trace_with_metrics_accepted_by_agent(self, metrics):
...
+            {"trace_id": 12345, "span_id": 67890},
+        ],
+    )
+    def test_trace_with_links_accepted_by_agent(self, span_links_kwargs):
+        """The agent does not return an error code when accepting a trace payload containing span links."""
+        tracer = Tracer()
+        with mock.patch("ddtrace.internal.writer.writer.log") as log:
+            with tracer.trace("root", service="test_encoding", resource="test_resource") as root:
+                root.set_link(**span_links_kwargs)
+                for _ in range(999):
```
Comment (erikayasuda) [inline tests/integration/test_encoding.py:79]: nit: Do we need this to iterate this many times? Not a huge deal, but wondering if there was a reason to make it generate that many traces

**Review by erikayasuda — APPROVED** [review]

File: tests/integration/test_encoding.py (hunk 2 above)
Comment (apiarian-datadog) [reply tests/integration/test_encoding.py:79]: probably not, i was just following the existing pattern

Comment (apiarian-datadog) [reply tests/integration/test_encoding.py:79]: i should probably rebase this so i'll change the number while i'm at it.

Comment (apiarian-datadog) [reply tests/integration/test_encoding.py]: changed it to 10

**Pull request merged**